src/FieldsService.php

```
@@ -0,0 +1,1533 @@
+            ->pluck('name')
+            ->toArray();
+    }
+
+    /**
+     * @return string[]
+     */
+    public function getRequiredFieldsWithNullables()
+    {
+        $this->throwIfNotUsingModelMethodFirst();
+
+        return $this->getRequiredFields($withNullables = true, $withDefaults = false, $withPrimaryKey = false);
```

PHPMD (2.15.0)

Avoid unused local variables such as '$withNullables'. (undefined)

(UnusedLocalVariable)